pdc_client/plugins/product.py

```diff
@@ -0,0 +1,145 @@
+# http://opensource.org/licenses/MIT
+#
+
+from __future__ import print_function
+
+import json
+
+from pdc_client.plugin_helpers import (PDCClientPlugin,
+                                       extract_arguments,
+                                       add_create_update_args)
+
+
+class ProductPlugin(PDCClientPlugin):
+    def register(self):
+        self.set_command('product')
+
+        list_parser = self.add_action('list', help='list all products')
+        self.add_filter_arguments(list_parser)
+        list_parser.set_defaults(func=self.list_products)
+
+        info_parser = self.add_action('info', help='display details of a product')
+        info_parser.add_argument('product_id', metavar='PRODUCT_ID')
+        info_parser.set_defaults(func=self.product_info)
+
+        update_parser = self.add_action('update', help='update an existing product')
+        update_parser.add_argument('product_id', metavar='PRODUCT_ID')
+        self.add_product_arguments(update_parser)
+        update_parser.set_defaults(func=self.product_update)
+
+        create_parser = self.add_action('create', help='create new product')
+        self.add_product_arguments(create_parser, required=True)
+        create_parser.set_defaults(func=self.product_create)
+
+    def add_product_arguments(self, parser, required=False):
+        required_args = {
+            'name': {},
+            'short': {}}
+
+        optional_args = {
+            'internal': {'action': 'store_const', 'const': True}}
+        add_create_update_args(parser, required_args, optional_args, required)
+
+        self.run_hook('product_parser_setup', parser)
+
+    def add_filter_arguments(self, parser):
+        active_group = parser.add_mutually_exclusive_group()
+        active_group.add_argument('--inactive', action='store_true', help='show only inactive products')
+        active_group.add_argument('--all', action='store_true', help='show both active and inactive products')
+
+        internal_group = parser.add_mutually_exclusive_group()
+        internal_group.add_argument('--internal', action='store_true', help='show only internal products')
+        internal_group.add_argument('--external', action='store_true', help='show only external products')
+
+        parser.add_argument('--name', help='show only products with matching name')
+        parser.add_argument('--short', help='show only products with matching short name')
+
+    def list_products(self, args):
+        filters = {}
+        if args.inactive:
+            filters['active'] = False
+        elif not args.all:
+            filters['active'] = True
+        if args.internal:
+            filters['internal'] = True
+        elif args.external:
+            filters['internal'] = False
+        if args.name:
+            filters['name'] = args.name
+        if args.short:
+            filters['short'] = args.short
+
+        products = list(self.client.get_paged(self.client.products._, **filters))
```

Comment:
When running this plugin with --page-size optional argument;  the results can't be return as expected.  
Maybe update this line: 
 ` products = self.client.get_paged(self.client.products._, **filters)`
And also need to  update the other codes as synic.